radix-engine/src/engine/modules/auth_module.rs

```diff
@@ -194,7 +197,25 @@ impl AuthModule {
                             system_api.lock_substate(node_id, offset, LockFlags::read_only())?;
                         let substate_ref = system_api.get_ref(handle)?;
                         let resource_manager = substate_ref.resource_manager();
-                        let auth = vec![resource_manager.get_vault_auth(*vault_fn).clone()];
+
+                        let auth = match visibility {
```

Comment:
Seems bad abstraction to me. Auth should be based on method rather than visibility origin, although I can see the motivation for resuing the same methods. :)

Reply:
Agreed it's a bit weird but struggled to find a better abstraction. Suggest to keep it simple like this for now but add a comment to continue looking for better abstraction?

Reply:
The current implementation is to make auth a function of **ref origin**, method, and access rules.

The alternative approach is to have dedicated methods for recall at the API. This introduces the question whether/how we want to limit "direct access" to a particular method, instead of any method. Yulong believed that we could rely on auth to do the right job, but ultimately it's a product question.

Will keep this topic in the background for future discussion.